orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandlerTest.kt

```diff
@@ -935,4 +920,47 @@ object CompleteStageHandlerTest : SubjectSpek<CompleteStageHandler>({
       }
     }
   }
+
+  setOf(TERMINAL).forEach { taskStatus ->
```

Comment:
Thoughts on whether we should perform the rollback before on `CANCEL`? 

I'm leaning towards yes.